docs/rum_collection.md

```diff
@@ -0,0 +1,271 @@
+```swift
+let session = URLSession(
+    configuration: .default,
+    delegate: DDURLSessionDelegate(),
+    delegateQueue: nil
+)
+```
+
+This will make the SDK track requests sent from this instance of the `URLSession`. Requests which URLs match the `firstPartyHosts` will be additionally marked as "first party" in the RUM Explorer.
+
+### RUM Actions
+
+To enable RUM actions tracking, use the `.trackUIKitActions(_:)` option when configuring the SDK:
+```
+Datadog.Configuration
+   .builderUsing(...)
+   .trackUIKitActions(true)
+   .build()
+```
+
+This will make the SDK track all significant taps occurring in the app. For privacy reasons, all interactions with the on-screen keyboard are ignored.
+
+### RUM Errors
+
+All "error" and "critical" logs will be reported as RUM errors and linked to the current RUM view:
+```swift
+logger.error("message")
+logger.critical("message")
+```
+
+Similarly, all ended APM spans marked as error will be reported as RUM errors:
+```swift
+span.setTag(key: OTTags.error, value: true)
+```
+
+## Manual Instrumentation
+
+### RUM Views
+
+Use the following methods on `Global.rum` to manually collect RUM resources:
+- `.startView(viewController:)`
+- `.stopView(viewController:)`
+
+Example:
+```swift
+// in your `UIViewController`:
+
+override func viewDidAppear(_ animated: Bool) {
+  super.viewDidAppear(animated)
+  Global.rum.startView(viewController: self)
+}
+
+override func viewDidDisappear(_ animated: Bool) {
+  super.viewDidDisappear(animated)
+  Global.rum.stopView(viewController: self)
+}
+```
+For more details and available options, please refer to the code documentation comments.
+
+### RUM Resources
+
+Use the following methods on `Global.rum` to manually collect RUM resources:
+* `.startResourceLoading(resourceName:url:httpMethod:)`
+* `.stopResourceLoading(resourceName:kind:)`
+* `.stopResourceLoadingWithError(resourceName:error:source:)`
+* `.stopResourceLoadingWithError(resourceName:errorMessage:source:)`
+
+Example:
+```swift
+// in your network client:
+
+Global.rum.startResourceLoading(
+    resourceName: "resource-name",
+    url: requestURL,
+    httpMethod: .GET
+)
+
+Global.rum.stopResourceLoading(
+    resourceName: "resource-name",
+    kind: .image,
+    httpStatusCode: 200
+)
+```
+
+**Note**: the `String` used for `resourceName` in both calls must be unique for the resource you are calling, so the SDK can match resource start with its completion. 
+
+For more details and available options, please refer to the code documentation comments.
+
+### RUM Actions
+
+To manually register RUM action, use either:
+* `.registerUserAction(type:name:)`
+
+or:
+* `.startUserAction(type:name:)`
+* and `.stopUserAction(type:)`
+
+on `Global.rum`. The first method can be used for sending actions which have no time (e.g. `.tap`), while the other two should be used for actions which define start and stop time (e.g. `.scroll`).
```

Comment:
```suggestion
on `Global.rum`. The first method can be used for sending actions that have no time (e.g. `.tap`), while the other two are intended for actions that define start and stop time (e.g. `.scroll`).
```

Reply:
This one was changed on request, while editing, sorry 🙂.